lib/Dialect/SMT/SMTOps.cpp

```diff
@@ -14,5 +14,40 @@ using namespace circt;
 using namespace smt;
 using namespace mlir;
 
+//===----------------------------------------------------------------------===//
+// BVConstantOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult BVConstantOp::verify() {
+  if (getValue().getType() != getType())
+    return emitError(
+        "smt.bv.constant attribute bitwidth doesn't match return type");
+
```

Comment:
Do you know why `FirstAttrDerivedResultType` is not catching this error?

Reply:
Excellent point. It actually does. However, it doesn't let me eliminate the custom `inferReturnTypes` because the assembly format verifier complains that the type is not buildable. Seems like an unnecessary restriction upstream.